tile/TileSeparateCrossbarRTL.py

```diff
@@ -186,23 +186,44 @@ def update_opt():
       s.routing_crossbar.recv_opt.msg @= s.ctrl_mem.send_ctrl.msg
       s.fu_crossbar.recv_opt.msg @= s.ctrl_mem.send_ctrl.msg
 
-      s.element.recv_opt.en @= s.ctrl_mem.send_ctrl.en
-      s.routing_crossbar.recv_opt.en @= s.ctrl_mem.send_ctrl.en
-      s.fu_crossbar.recv_opt.en @= s.ctrl_mem.send_ctrl.en
-
-      s.ctrl_mem.send_ctrl.rdy @= s.element.recv_opt.rdy & \
-                                  s.routing_crossbar.recv_opt.rdy & \
-                                  s.fu_crossbar.recv_opt.rdy
-
+      s.element.recv_opt.val @= s.ctrl_mem.send_ctrl.val & ~s.element_done
+      s.routing_crossbar.recv_opt.val @= s.ctrl_mem.send_ctrl.val & ~s.routing_crossbar_done
+      s.fu_crossbar.recv_opt.val @= s.ctrl_mem.send_ctrl.val & ~s.fu_crossbar_done
+
+      # s.ctrl_mem.send_ctrl.rdy @= s.element.recv_opt.rdy & \
+      #                             s.routing_crossbar.recv_opt.rdy & \
+      #                             s.fu_crossbar.recv_opt.rdy
+
+      # FIXME: yo96, rename ctrl.rdy to ctrl.proceed or sth similar.
+      # Allows either the FU-related go out first or routing-xbar go out first. And only
+      # allows the ctrl signal proceed till all the sub-modules done their own job (once).
+      s.ctrl_mem.send_ctrl.rdy @= (s.element.recv_opt.rdy | s.element_done) & \
+                                  (s.routing_crossbar.recv_opt.rdy | s.routing_crossbar_done) & \
+                                  (s.fu_crossbar.recv_opt.rdy | s.fu_crossbar_done)
+
+    # Updates the signals indicating whether certain modules already done their jobs.
+    @update_ff
+    def already_done():
+      if s.reset | s.ctrl_mem.send_ctrl.rdy:

```

Comment:
@yo96 We don't need `(s.ctrl_mem.send_ctrl.rdy == b1(1))`, right?

Reply:
Correct. We should avoid writing things like` == b1(1)` or `== b1(0)` to make the code more compact and readable.